client/packages/common/src/hooks/useUrlQuery/useUrlQueryParams.ts

```diff
@@ -76,8 +76,8 @@ export const useUrlQueryParams = ({
     offset: urlQuery.page ? (urlQuery.page - 1) * rowsPerPage : 0,
     first: rowsPerPage,
     sortBy: {
-      key: urlQuery.sort ?? initialSort,
-      direction: urlQuery.dir ?? 'asc',
+      key: urlQuery.sort ?? initialSort?.key,
```

Comment:
annoying that the UrlQuery hooks are using 'dir' and the sort is using 'direction'. Was hoping we could just use `sortBy: initialSort ?? { ... `

anyhoo.. 
would you mind updating line `39` to 
```
    updateQuery({ sort, dir: dir === 'desc' ? 'desc' : '' });
```

that makes the url query shorter and more consistent.

Reply:
> I have also updated the initial sort in `usePatients` to be by `lastName` as I don't find code that helpful shrug

Sounds good, did you do it in this PR? no commit here...

Reply:
thanks good ideas, I made these changes

Reply:
oh no!
@clemens-msupply I have found a problem.. this has broken our column sort. If you try to change direction on a column, it won't work.

The problem is that when sorting ascending, we remove the `dir` parameter
Then, in this line

```
 direction: urlQuery.dir ?? initialSort?.dir ?? 'asc',
```

the `urlQuery.dir` will return undefined, and so will always return the `initialSort?.dir`. If the initial sort has been set, then you are unable to change direction!

I've put a suggestion on [this PR](https://github.com/openmsupply/open-msupply/pull/2237) and tested the initial sort, and that's working for me. Can you have a quick check as well? afaics the `useEffect` should be handling the initial sort without the need for this change